docs/src/pages/migration-guide.md

```diff
@@ -0,0 +1,221 @@
+
+```astro
+---
+// tick.astro
+import { green } from '@radix-ui/colors'
+---
+<style define:vars={green}>
+  h-tick {
+    background-color: var(--green4);
+    border-radius: 50%;
+    color: var(--green11);
+    height: 15px;
+    width: 15px;
+  }
+</style>
+<h-tick>✓</h-tick>
+```
+
+
+
+## Components in Markdown
+
+In Astro v0.21, Components can be used within Markdown files.
+
+```md
+---
+Layout: '...'
+setup: | 
+  import MyReactComponent from '../components/MyReactComponent.jsx'
+---
+
+# Hydrating on visibility
+
+<MyReactComponent client:visible />
+```
+
+
+
+## Environment variables
+
+In Astro v0.21, environment variables are loaded from `.env` files in your project directory.
+
+```ini
+.env                # loaded in all cases
+.env.local          # loaded in all cases, ignored by git
+.env.[mode]         # only loaded in specified mode
+.env.[mode].local   # only loaded in specified mode, ignored by git
+```
+
+For security purposes, only variables prefixed with `PUBLIC_` are accessible to your code.
+In prior releases, these variables were prefixed with `SNOWPACK_PUBLIC_` and required the `@snowpack/plugin-env` plugin.
+
+**Example**
+
+```ini
+SECRET_PASSWORD=password123
+VITE_ANYBODY=there
+```
+
+In that example, only `VITE_ANYBODY` will be exposed as `import.meta.env.VITE_ANYBODY` to your client source code, but `SECRET_PASSWORD` will not.
+
+
+
+## File references
+
+In Astro v0.21, files need to be referenced with their extension exactly as it is on disk.
+
+This means that `file.tsx` files need to be referenced as `file.tsx` and not as `file.jsx`.
+
+**Example**
+
+```tsx
+// Div.tsx
+export default function Div(props) {
+  return <div />
+}
+```
+
+```diff
+- import Div from './Div.jsx' // Astro v0.20
++ import Div from './Div.tsx' // Astro v0.21
+```
+
+```scss
+// Div.scss
+div {
+  all: unset
+}
+```
+
+```diff
+- <link rel="stylesheet" href="./Div.css">
++ <link rel="stylesheet" href="./Div.scss">
+```
+
+
+
+## Applying plugins
+
+In Astro v0.21, plugins are configured in `vite.plugins`.
+
+```js
+import { imagetools } from 'vite-imagetools'
+
+export default {
+  vite: {
+    plugins: [
+      imagetools()
+    ]
+  }
+}
+```
+
+To learn more about Vite plugins, please visit their [plugin guide](https://vitejs.dev/guide/using-plugins.html).
+
+
+
+## Writing plugins
+
+In Astro v0.21, plugins previously using `snowpackPlugin` or `snowpackPluginOptions` should now use `viteConfig()`.
+
+```diff
+// @astrojs/renderer-svelte/index.js
++ import { svelte } from '@sveltejs/vite-plugin-svelte';
+
+export default {
+  name: '@astrojs/renderer-svelte',
+  client: './client.js',
+  server: './server.js',
+-  snowpackPlugin: '@snowpack/plugin-svelte',
+-  snowpackPluginOptions: { compilerOptions: { hydratable: true } },
++  viteConfig() {
++    return {
++      optimizeDeps: {
++        include: ['@astrojs/renderer-svelte/client.js', 'svelte', 'svelte/internal'],
```

Comment:
Minor suggestion: should we have a different example? Renderers should set this up themselves and I’d like people to not have to think they have to manage this themselves (it can be frustrating).

Also, if there’s an issue with the renderers I’d like to know so that users don’t have to manage this!

Reply:
Ah sorry; nevermind. I missed the heading and thought I was still in the generic user config as above.